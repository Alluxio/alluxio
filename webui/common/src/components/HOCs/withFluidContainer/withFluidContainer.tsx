/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import React from 'react';
import {getDisplayName} from "../../../utilities";

export interface IFluidContainerProps {
    class: string;
}

export function withFluidContainer<T extends IFluidContainerProps>(WrappedComponent: React.ComponentType<T>) {
    const fluidContainerHoc = (props: T) => {
        return (
            <div className={props.class}>
                <div className="container-fluid">
                    <div className="row">
                        <WrappedComponent {...props} />
                    </div>
                </div>
            </div>
        )
    };
    (fluidContainerHoc as React.FunctionComponent).displayName = `withFluidContainer(${getDisplayName(WrappedComponent)})`;
    return fluidContainerHoc;
}
