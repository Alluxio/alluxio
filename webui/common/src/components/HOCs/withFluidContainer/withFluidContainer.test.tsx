/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {configure, shallow, ShallowWrapper} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import {IFluidContainerProps, withFluidContainer} from "./withFluidContainer";

configure({adapter: new Adapter()});

const WrappedComponent = () => <div>Wrapped</div>;
const EnhancedComponent = withFluidContainer(WrappedComponent);

describe('withLoadingMessage HOC', () => {
    let props: IFluidContainerProps;

    beforeAll(() => {
        props = {
            class: 'test'
        };
    });

    describe('Shallow component', () => {
        let shallowWrapper: ShallowWrapper;

        beforeAll(() => {
            shallowWrapper = shallow(<EnhancedComponent {...props}/>);
        });

        it('Renders without crashing', () => {
            expect(shallowWrapper.length).toEqual(1);
        });

        it('Contains a div with class test', () => {
            expect(shallowWrapper.find('.test').length).toEqual(1);
        });

        it('Contains a div with class container-fluid', () => {
            expect(shallowWrapper.find('.container-fluid').length).toEqual(1);
        });

        it('Contains a div with class row', () => {
            expect(shallowWrapper.find('.row').length).toEqual(1);
        });

        it('Matches snapshot - renders LoadingMessage', () => {
            expect(shallowWrapper).toMatchSnapshot();
        });
    });
});
