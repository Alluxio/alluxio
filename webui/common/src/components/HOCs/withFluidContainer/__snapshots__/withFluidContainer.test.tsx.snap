// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLoadingMessage HOC Shallow component Matches snapshot - renders LoadingMessage 1`] = `
<div
  className="test"
>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <WrappedComponent
        class="test"
      />
    </div>
  </div>
</div>
`;
